src/backend/src/controllers/work-packages.controllers.ts

```diff
@@ -156,12 +162,16 @@ export const createWorkPackage = async (req: Request, res: Response) => {
     })
   );
 
-  const dependenciesIds = dependenciesWBSElems.map((elem) => {
-    if (elem === null) throw new TypeError('One of the dependencies was not found.');
-    return elem.wbsElementId;
+  const dependenciesIds: number[] = [];
+  // populate dependenciesIds with the element ID's
+  // and return error 400 if any elems are null
+  dependenciesWBSElems.forEach((elem) => {
+    if (elem === null) {
+      return res.status(400).json({ message: 'One of the dependencies was not found.' });
```

Comment:
this won't work because returning inside a forEach just ends the foreach and doesn't actually return from the outer function. set up a boolean outside like `dependenciesHasNulls` or something then after the forEach if it's true then you return the res 400

Reply:
Why would we want to run the entire foreach loop if there's a null?

Reply:
you're right, what I'm saying is that returning inside a foreach only ends the foreach (it's like `break`). the outer endpoint will keep running after that. You should end early if you see a null, but you also need a way to make sure the outer endpoint finishes too if you do that